internal/processor/parser/app.go

```diff
@@ -257,6 +176,104 @@ func (am *AppParser) Parse(envelope *events.Envelope) ([]metric.MetricPackage, e
 	return metricsPackages, nil
 }
 
+// listApps is meant to replace the function from the go-cfclient and allow fetching apps in parallel tasks
+// The apps returned by this are just missing a reference to the client (unexported property) so don't use the Space() and Summary() methods
+func listApps(c *cfclient.Client, numWorkers int, log *gosteno.Logger) ([]cfclient.App, error) {
+
+	// Query the first page to get the total number of pages.
+	resp, err := getAppsPage(c, 1)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error requesting apps page 1, skipping cache warmup")
+	}
+	appResources := resp.Resources
+
+	var mutex sync.Mutex
+	var wg sync.WaitGroup
+
+	// Page 1 already fetched
+	pages := resp.Pages - 1
+	// No need for more workers than pages left
+	numWorkers = int(math.Min(float64(numWorkers), float64(pages)))
+	if pages > 0 {
```

Comment:
I would prefer not having everything inside this condition. Just the `pagesPerWorker` logic. Would that be a problem regarding `wg.Wait()`. Also, let's add a test for this case.